cob_twist_controller/src/limiters/limiter.cpp

```diff
@@ -320,3 +356,40 @@ KDL::JntArray LimiterIndividualJointAccelerations::enforceLimits(const KDL::JntA
     return q_dot_norm;
 }
 /* END LimiterIndividualJointAccelerations **********************************************************************/
+
+/* BEGIN LimiterCartesianVelocities ********************************************************************/
+/**
+ * This implementation implements a saturation function to the Cartesian twists
+ */
+KDL::Twist LimiterCartesianVelocities::enforceCartesianLimits(const KDL::Twist& v_in) const
+{
+    KDL::Twist v_out(v_in);
+    double factor = 1.0;
+    // linear limts
+    for (unsigned int i = 0; i < 3; i++)
```

Comment:
This is such complex function to just to a simple thing... I think that just makes the code even harder for future developers.. But I will put it..